library/core/src/ptr/non_null.rs

```diff
@@ -1828,9 +1828,8 @@ impl<T: ?Sized> hash::Hash for NonNull<T> {
 impl<T: ?Sized> From<Unique<T>> for NonNull<T> {
     #[inline]
     fn from(unique: Unique<T>) -> Self {
-        // SAFETY: A Unique pointer cannot be null, so the conditions for
-        // new_unchecked() are respected.
-        unsafe { NonNull::new_unchecked(unique.as_ptr()) }
+        // SAFETY: A Unique pointer cannot be null.
+        unsafe { NonNull { pointer: unique.as_ptr() } }
```

Comment:
Since `Unique` is unstable, maybe just change `as_ptr` to return the inner `NonNull`

Reply:
Hm, that would be unlike our other `as_ptr` methods that all return raw pointers, and this is used in a handful of place that do need a `*mut T`, so they'd have to be `.as_ptr().as_ptr()`. `NonNull` already has an `as_non_null_ptr` so I suppose there's some precedent at least?